guides/plugins/plugins/administration/add-rule-assignment-configuration.md

```diff
@@ -0,0 +1,256 @@
+            associationEntitiesConfig.push(/* insert your configuration here */);
+            return associationEntitiesConfig;
+        },
+    }
+});
+```
+{% endcode %}
+
+## Adding the configuration
+
+The configuration of the rule assignment is an object and offers a wide range of configuration options.
+Just have a look onto one example configuration item of the `Dynamic Access` plugin:
+
+{% code title="Example of a configuration item" %}
+```javascript
+getRuleAssignmentConfig()
+{
+    return [
+        {
+            id: 'swagDynamicAccessProducts',
+            notAssignedDataTotal: 0,
+            entityName: 'product',
+            label: 'swag-dynamic-access.sw-settings-rule.detail.associations.productVisibility',
+            criteria: () => {
+                const criteria = new Criteria();
+                criteria.setLimit(this.associationLimit);
+                criteria.addFilter(Criteria.equals('swagDynamicAccessRules.id', this.rule.id));
+                criteria.addAssociation('options.group');
+                criteria.addAssociation('swagDynamicAccessRules');
+
+                return criteria;
+            },
+            api: () => {
+                const api = Object.assign({}, Context.api);
+                api.inheritance = true;
+
+                return api;
+            },
+            detailRoute: 'sw.product.detail.base',
+            gridColumns: [
+                {
+                    property: 'name',
+                    label: 'Name',
+                    rawData: true,
+                    sortable: true,
+                    routerLink: 'sw.product.detail.prices',
+                    allowEdit: false,
+                },
+            ],
+            deleteContext: {
+                type: 'many-to-many',
+                entity: 'product',
+                column: 'extensions.swagDynamicAccessRules',
+            },
+            addContext: {
+                type: 'many-to-many',
+                entity: 'swag_dynamic_access_product_rule',
+                column: 'productId',
+                searchColumn: 'name',
+                criteria: () => {
+                    const criteria = new Criteria();
+                    criteria.addFilter(
+                            Criteria.not('AND', [Criteria.equals('swagDynamicAccessRules.id', this.rule.id)]),
+                    );
+                    criteria.addAssociation('options.group');
+
+                    return criteria;
+                },
+                gridColumns: [
+                    {
+                        property: 'name',
+                        label: 'Name',
+                        rawData: true,
+                        sortable: true,
+                        allowEdit: false,
+                    },
+                    // ...
+                ],
+            },
+        },
+    ];
+}
+```
+{% endcode %}
+
+Let's go through the most important entries, how to configure your rule assignment.
```

Comment:
Maybe this is more readable if you create a small table like. Technically, this is already a "reference", but I think for this small case it's fine to just have it right here in the guide, where it's needed.

| Option | Description |
| --- | --- |
| **id** | Required identifier for the assignment, which is arbitrary but unique |
| **entityName, criteria, id** | Required for data loading |